### PR TITLE
Add `copy()` method for `DiffractionObject`

### LINKED ISSUE #201: add a copy method to diffraction objects
### Problem
That would be nice.  Something like 
```
def copy(self):
    return DiffractionObject(**self.everything)
```
where teat this as pseudocode....it is supposed to show intent, not how it would actually be coded.

usage would be that a test like this would pass:
```
copy_of_DO = DO.copy()
assert copy_of_DO == DO
```

### PR DESCRIPTION
**Pull request by bobleesj**
Closes #201 - add `copy()` for `DiffracitonObject`

### CODE REVIEW

File: tests/test_diffraction_objects.py [hunk 1]
```diff
@@ -410,3 +410,15 @@ def test_all_array_setter():
...
+def test_copy_object():
+    do = DiffractionObject(
+        name="test",
+        wavelength=4.0 * np.pi,
+        xarray=np.array([0.0, 90.0, 180.0]),
+        yarray=np.array([1.0, 2.0, 3.0]),
+        xtype="tth",
+    )
+    copy_of_DO = do.copy()
+    assert do == copy_of_DO
```
Comment (bobleesj) [inline tests/test_diffraction_objects.py:424]: Utizling the object's `__eq__` nice implementation

**Comment by bobleesj** [comment]
@sbillinge ready for review

File: tests/test_diffraction_objects.py (hunk 1 above)
Comment (sbillinge) [reply tests/test_diffraction_objects.py:424]: To complete this test also assert id(actual) != Id(initial)

Comment (sbillinge) [reply tests/test_diffraction_objects.py:424]: Then the copy is not pointing to the same object

**Review by sbillinge — COMMENTED** [review]
Pls see inline

**Comment by bobleesj** [comment]
@sbillinge Right! Thank you. Additional test added.

File: tests/test_diffraction_objects.py (hunk 1 above)
Comment (bobleesj) [reply tests/test_diffraction_objects.py:424]: yes done below.

**Comment by bobleesj** [comment]
@sbillinge ready for review - just enhanced the variable name from `copy_of_DO` to `do_copy` for consistency.

**Comment by sbillinge** [comment]
@bobleesj not sure why codecov is complaining here...diffpy has a global org secret token.  Maybe it is being flaky.  I will just try and rerun it?  after than an empty commit?

**Comment by bobleesj** [comment]
Screenshot 2024-12-08 at 10 05 26 AM

Codecov looks okay to me in fact 0.05% has gone up! @sbillinge

**Comment by sbillinge** [comment]
> Screenshot 2024-12-08 at 10 05 26 AM
> 
> Codecov looks okay to me in fact 0.05% has gone up! @sbillinge

it failed to upload so failed.  I reran it and it uploaded and passed.....   CC is officially flaky....

**Pull request merged**